retrieverdash/dashboard_script/dashboard_script.py

```diff
@@ -13,24 +14,26 @@
 from .status_dashboard_tools import create_dirs
 from .status_dashboard_tools import dataset_type
 
-file_location = os.path.dirname(os.path.realpath(__file__))
+file_location = os.path.normpath(os.path.dirname(os.path.realpath(__file__)))
+
+IGNORE_LIST = ['prism-climate', 'mammal-super-tree', 'forest-inventory-analysis',
+               'vertnet']
 
-IGNORE_LIST = ['prism-climate', 'mammal-super-tree']
 
 def check_dataset(dataset):
-    os.chdir(os.path.join(file_location))
+    os.chdir(file_location)
     md5 = None
     status = None
     reason = None
     diff = None
+    dataset_detail = None
     try:
         try:
-            dataset_detail = json.load(open('dataset_details.json', 'r'))
-        except FileNotFoundError:
-            with open("dataset_details.json", 'w') as json_file:
-                dataset_detail = dict()
-                dataset_detail['dataset_details'] = {}
-                json.dump(dataset_detail, json_file)
+            with open("dataset_details.json", 'r') as json_file:
+                dataset_detail = json.load(json_file)
```

Comment:
This json file gives us the details of all the status for each dataset.
How about we read it in once, before we even enter this function.
Then we pass the data as a dictionary.

When we finish all the precessing of all datasets, the dictionary is written back to the json

Reply:
The current code is a bit according to the case in which we use multiple processes. I will modify it according to a single thread.

Reply:
In the current way, we can see the dataset_details.json getting updated after each dataset check.
What would happen if the files are changed during the execution of the script and someone opens the dashboard and tries to access those files?
I think we should write to the dataset_details.json after each dataset is checked.